### PR TITLE
[ci:component:github.com/gardener/machine-controller-manager:v0.49.1->v0.51.1]

### PR DESCRIPTION
**Pull request by gardener-robot-ci-1**
**Release Notes**:
```other operator github.com/gardener/machine-controller-manager #842 @unmarshall
New metrics introduced: 
- api_request_duration_seconds -> tracks time taken for successful invocation of provider APIs. This metric can be filtered by provider and service.
- driver_request_duration_seconds -> tracks total time taken to successfully complete driver method invocation. This metric can be filtered by provider and operation.
- driver_requests_failed_total -> records total number of failed driver API requests. This metric can be filtered by provider, operations and error_code.
```
```other developer github.com/gardener/machine-controller-manager #845 @unmarshall
A new make target is introduced to add license headers.
```
```other developer github.com/gardener/machine-controller-manager #823 @himanshu-kun
Removed dead metrics code and refactored the remaining metrics code
```
```other operator github.com/gardener/machine-controller-manager #851 @rishabh-11
Added `errorCode` field in the `LastOperation` struct. This should be implemented only for the `CreateMachine` call in the `triggerCreationFlow`. This field will be utilized by Cluster autoscaler to do early backoff 
```
```bugfix operator github.com/gardener/machine-controller-manager #833 @rishabh-11
Included `UnavailableReplicas` in determining if a machine deployment status update is needed
```
```other operator github.com/gardener/machine-controller-manager #866 @elankath
The default `machine-safety-orphan-vms-period` has been reduced from 30m to 15m.
```
```breaking operator github.com/gardener/machine-controller-manager #878 @ccwienk
Change OCI Image Registry from GCR (`eu.gcr.io/gardener-project`) to Artifact-Registry (`europe-docker.pkg.dev/gardener-project/releases`). Users should update their references.
```
```other operator github.com/gardener/machine-controller-manager #869 @piyuagr
fixed IT for seed with k8s >= 1.27 as control cluster 
```
```bugfix operator github.com/gardener/machine-controller-manager #887 @elankath
Fix for edge case of Node object deletion missed during machine termination.
```
```other operator github.com/gardener/machine-controller-manager #877 @himanshu-kun
machine controller won't reconcile machine on non-spec update events
```
```bugfix operator github.com/gardener/machine-controller-manager #839 @elankath
Force drain and delete volume attachments for nodes un-healthy due to `ReadOnlyFileSystem` and `NotReady` for too long
```
```other operator github.com/gardener/machine-controller-manager #808 @jguipi
Added a new metric that will allow to get the number of stale (due to unhealthiness) machines  that are getting terminated
```
```doc developer github.com/gardener/machine-controller-manager #886 @himanshu-kun
Phase transition diagram for a machine object is added to FAQs
```
```bugfix developer github.com/gardener/machine-controller-manager #879 @sssash18
MCM restart happens properly in integration tests now. This fix will get activated, once this version is vendored in your mcm-provider
```
```other developer github.com/gardener/machine-controller-manager #858 @afritzler
Bump `k8s.io/*` deps to `v0.28.2`
```
```bugfix user github.com/gardener/machine-controller-manager #821 @rishabh-11
An edge case where outdated DesiredReplicas annotation blocked a rolling update is fixed.
```
```other operator github.com/gardener/machine-controller-manager #827 @rishabh-11
Updated to go v1.20.5
```
```other operator github.com/gardener/machine-controller-manager #852 @unmarshall
Makefile targets have changed: Introduced gardener-setup, gardener-restore, gardener-local-mcm-up, non-gardener-setup, non-gardener-restore,  non-gardener-local-mcm-up. Users can also directly use the scripts which are used by these makefile targets.
```
```other developer github.com/gardener/machine-controller-manager #820 @afritzler
Bump `k8s.io/*` deps to v0.27.2
```
```other developer github.com/gardener/machine-controller-manager #842 @unmarshall
status.Status now captures underline cause, allowing consumers to introspect the error returned by the provider. WrapError() function could be used to wrap the provider error
```
```bugfix operator github.com/gardener/machine-controller-manager #866 @elankath
Removes `node.machine.sapcloud.io/not-managed-by-mcm` annotation from nodes managed by the MCM.
```

### CODE REVIEW

File: .ci/pipeline_integration_test [hunk 1]
```diff
@@ -74,25 +84,93 @@ function hf_wait_on() {
...
+
+function fetch_userdata(){
+    NAMESPACE='garden'
+    kubectl --kubeconfig=dev/sa_kubeconfig.yaml get shoot $TEST_SHOOT -oyaml > dev/shoot.yaml
+    SEED_NAME=$(yq eval '.status.seedName' dev/shoot.yaml)
+    request_kubeconfig $NAMESPACE $SEED_NAME "dev/seed_kubeconfig.yaml"
+    USERDATA_SECRET_NAME=$(kubectl --kubeconfig=dev/seed_kubeconfig.yaml get secrets -n shoot--mcm-ci--$TEST_SHOOT -o custom-columns=Name:.metadata.name --no-headers | grep shoot--mcm-ci)
+    kubectl --kubeconfig=dev/seed_kubeconfig.yaml get secret $USERDATA_SECRET_NAME -n shoot--mcm-ci--$TEST_SHOOT -oyaml > dev/userData.yaml
+    userData=$(yq eval '.data.userData' dev/userData.yaml)
+    echo $userData
```
Comment (rishabh-11) [inline .ci/pipeline_integration_test:117]:
```suggestion
```

**Review by rishabh-11 — CHANGES_REQUESTED** [review]
Minor change needed, rest looks good

File: .ci/pipeline_integration_test [hunk 2]
```diff
@@ -74,25 +84,92 @@ function hf_wait_on() {
...
-    #convert json to yaml, so that machineclass can be parsed during integration-test
-    yq e -P ${SOURCE_PATH}/dev/v1machineclass.json > ${SOURCE_PATH}/dev/v1machineclass_converted.yaml
+function fetch_credentials(){
+   kubectl --kubeconfig=dev/sa_kubeconfig.yaml get secret $CREDENTIALS_SECRET_NAME -n $GARDEN_CORE_NAMESPACE -o yaml > dev/credentials.yaml
+   clientID=$(yq eval '.data.clientID' dev/credentials.yaml)
+   clientSecret=$(yq eval '.data.clientSecret' dev/credentials.yaml)
+   subscriptionID=$(yq eval '.data.subscriptionID' dev/credentials.yaml)
+   tenantID=$(yq eval '.data.tenantID' dev/credentials.yaml)
+
+   echo "$clientID" "$clientSecret" "$subscriptionID" "$tenantID"
```
Comment (rishabh-11) [inline .ci/pipeline_integration_test:133]:
```suggestion
```

**Review by rishabh-11 — CHANGES_REQUESTED** [review]

File: .ci/pipeline_integration_test (hunk 2 above)
Comment (elankath) [reply .ci/pipeline_integration_test:133]: sorry, had added these for debugging during auth failures.

File: .ci/pipeline_integration_test [hunk 3]
```diff
@@ -148,10 +238,19 @@ function check_cluster_state() {
...
 function run_integration_tests() {
+    echo "Printing env.."
+    env
+    echo "---"
+    echo "--- Printing CONTROL_KUBECONFIG.."
+    cat $CONTROL_KUBECONFIG
+    echo "---"
+    echo "--- Printing TARGET_KUBECONFIG..."
+    cat $TARGET_KUBECONFIG
+    echo "---"
```
Comment (rishabh-11) [inline .ci/pipeline_integration_test:249]:
```suggestion
```
We should not print the kubeconfig

**Review by rishabh-11 — CHANGES_REQUESTED** [review]

**Review by rishabh-11 — APPROVED** [review]
/lgtm

**Pull request merged**